src/AIAPI/MainGame/GameExtensions.cs

```diff
@@ -76,7 +76,7 @@ public static AgentActor GetNPC(this AgentData agentData)
 			{
 				if (agentTable.TryGetValue(key, out AgentActor agentActor))
 				{
-                    if (agentActor?.charaID == agentData.param.charaID) return agentActor;
+                    if (agentActor?.ChaControl.name == agentData.param?.actor.ChaControl.name) return agentActor;
```

Comment:
Why not compare `agentActor == agentData.param?.actor` or the ChaControls instead of names? Are you sure the name comparison is safe?

Reply:
changed to compare actor objects